### DISCUSSION TITLE
Can somebody explain to me how it possible implement Request-Reply pattern by this lib?

### DISCUSSION THREAD

**Discussion by gallun320** [nats-io/nats.net.v1#529] [Q&A]
Documentation does not provide information about how to respond to the request, and the implementation of Msg class really confused because the Respond method publish(!!!) by the topic but the request subscribe on the same topic as a subscriber and will got this message first, so I don't see any possibility to Respond on Request

**Comment by sixlettervariables** [comment]
You would subscribe to the request subject and reply to the subject provided in the message received.

You can check out our tests for examples of these.

**Comment by ColinSullivan1** [comment]
To expand a bit on @sixlettervariables, there is a [Request](https://nats-io.github.io/nats.net/interface_n_a_t_s_1_1_client_1_1_i_connection.html#a4210238627f16c565d8ce88421ada056) API that essentially publishes and waits for a response on a unique subject.  The responder subscribes to a subject, and when it receives a message, it can either publish on the [reply subject](https://nats-io.github.io/nats.net/class_n_a_t_s_1_1_client_1_1_msg.html#a1e18574e6eba1ea0a8540aff37f70732), or use a convenience method to reply with [msg.Respond()](https://nats-io.github.io/nats.net/class_n_a_t_s_1_1_client_1_1_msg.html#a656c04f814d8c3631a44aa4eeaf3de82)

The examples demonstrate this:
https://github.com/nats-io/nats.net/tree/master/src/Samples/Requestor
https://github.com/nats-io/nats.net/tree/master/src/Samples/Replier

I'd suggest walking through the examples.  There's more on request/reply here:  https://docs.nats.io/developing-with-nats/sending/request_reply
